apis/root_handler_test.go

```diff
@@ -51,7 +56,32 @@ func testRootAPI(t *testing.T, when spec.G, it spec.S) {
 	})
 
 	it("matches the expected response body format", func() {
-		expectedBody := `{"links":{"self":{"href":"` + defaultServerURL + `"},"bits_service":null,"cloud_controller_v2":null,"cloud_controller_v3":{"href":"` + defaultServerURL + `/v3","meta":{"version":"3.90.0"}},"network_policy_v0":null,"network_policy_v1":null,"login":null,"uaa":null,"credhub":null,"routing":null,"logging":null,"log_cache":null,"log_stream":null,"app_ssh":null}}`
-		g.Expect(rr.Body.String()).To(Equal(expectedBody), "Response body matches RootV3GetHandler response:")
+		var resp presenter.RootResponse
+		g.Expect(json.Unmarshal(rr.Body.Bytes(), &resp)).To(Succeed())
+
+		g.Expect(resp).To(gstruct.MatchAllFields(gstruct.Fields{
```

Comment:
We've had some discussion about making tests intentionally brittle when checking responses. To that end, the raw string comparison might be best here. The problem we noted with using the implementation code structs is that if the struct is incorrectly structured, the test will not surface this, while a raw string is less tolerant to this.

Reply:
In this case since it's matching fields against an explicit `gstruct.Fields` I think we're getting more coverage than what bit us in the past (expecting `actualStruct` to equal `testStruct`). Maybe this is missing some coverage of some of the JSON marshalling aspects, though.

Reply:
Sounds fine, then.